pkg/scheduler/algorithm/priorities/resource_allocation.go

```diff
@@ -44,44 +52,38 @@ func (r *ResourceAllocationPriority) PriorityMap(
 	if node == nil {
 		return schedulerapi.HostPriority{}, fmt.Errorf("node not found")
 	}
-	allocatable := nodeInfo.AllocatableResource()
-
-	var requested schedulernodeinfo.Resource
-	if priorityMeta, ok := meta.(*priorityMetadata); ok {
-		requested = *priorityMeta.nonZeroRequest
-	} else {
-		// We couldn't parse metadata - fallback to computing it.
-		requested = *getNonZeroRequests(pod)
+	if r.ResourceToWeightMap == nil {
+		return schedulerapi.HostPriority{}, fmt.Errorf("resources not found")
+	}
+	requested := make(ResourceToValueMap, len(r.ResourceToWeightMap))
```

Comment:
This used to be cached in the priorityMetadata. 

@Huang-Wei @bsalamat I am not sure why this was cached, pods usually have one or two containers, also  it seems to me that this function might be called more than once only if multiple resource-based priority functions are configured, is that actually common?

Reply:
I think this was cached because this function is called repeatedly for the same pod over multiple nodes. I wonder if we can do the same? basically create a function that returns the requested resources that we store in the metadata as [nonZeroRequest](https://github.com/kubernetes/kubernetes/blob/df0a99ad354e5e458a123a2ba5f8bd95c34660c0/pkg/scheduler/algorithm/priorities/metadata.go#L48:2) (of course we need to change the type to ResourceToValueMap)

Reply:
that being said, I am fine if we don't cache it since pods have very few containers anyways.